### PR TITLE
Add filter to list repository pipelines API

### PR DESCRIPTION
**Pull request by 6543**
we currently filter thing on the client side:

https://github.com/woodpecker-ci/woodpecker/blob/ce8cd029b4cbc9035bd4572c2b30af5bd128a079/web/src/views/repo/RepoPullRequest.vue#L27-L40

this makes it possible to filter server side in an upcomming pull addressing the frontend

### CODE REVIEW

**Comment by anbraten** [comment]
We still need to filter on the client though to get reactive queries updating on changes coming from the sse stream.

**Comment by 6543** [comment]
:point_up: well yes but we also need it from the server as getting a random count of entry's even if there should exist more ... is bad

e.g: https://ci.woodpecker-ci.org/repos/3780/branches/main

-> somethimes it just displays 2 ... or 3 ... or some other random number

**Comment by xoxys** [comment]
> but we have https://ci.woodpecker-ci.org/repos/3780/pipeline/15602 ....

Can you explain how server-side filtering help in this case?

**Comment by 6543** [comment]
well you currently can not discover it if you dont have the link or are on the specific pull to retrive the link

**Comment by xoxys** [comment]
That was clear already, I still don't understand how server-side filtering will solve it.

**Comment by 6543** [comment]
you have to query the server again and use that filter if you switch the pipeline list view to a different one

**Comment by 6543** [comment]
If I have time I'll also update the frontend within this pull at some point

File: server/api/pipeline.go [hunk 1]
```diff
@@ -116,14 +117,32 @@ func createTmpPipeline(event model.WebhookEvent, commit *model.Commit, user *mod
 //	@Param			perPage			query	int		false	"for response pagination, max items per page"	default(50)
 //	@Param			before			query	string	false	"only return pipelines before this RFC3339 date"
 //	@Param			after			query	string	false	"only return pipelines after this RFC3339 date"
+//	@Param			branch			query	string	false	"filter pipelines by branch"
+//	@Param			event			query	string	false	"filter pipelines by webhook events (comma separated)"
+//	@Param			ref				query	string	false	"filter pipelines by strings contained in ref"
```
Comment (xoxys) [inline server/api/pipeline.go:122]: What's the use case for the "ref contains" filter?

**Review by xoxys — APPROVED** [review]

**Pull request merged**